### PR TITLE
Bonnie 1090

### PR DESCRIPTION
**Pull request by lukoktonos**
Originally, only direct reference codes from the main library were added to the data criteria section.

With the most recent change, no direct reference codes were added to the "terminology section", which replaced the "data criteria" section.

With this change, all direct reference codes will be added to the "terminology" section.
Pull requests into Bonnie require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [X] This pull request describes why these changes were made.
- [X] JIRA ticket for this PR: https://jira.mitre.org/browse/BONNIE-1090
- [x] JIRA ticket links to this PR 
- [X] Code diff has been done and been reviewed (it **does not** contain: additional white space, not applicable code changes, debug statements, etc.)
- [X] If UI changes have been made, google WAVE plug-in has been executed to ensure no 508 issues were introduced.
- [X] Tests are included and test edge cases
- [X] Tests have been run locally and pass (remember to update Gemfile when applicable)
- [X] Code coverage has not gone down and all code touched or added is covered. 
     * See [front-end testing instructions](https://github.com/projecttacoma/bonnie/wiki/Testing#frontend-testing) for getting coverage for front-end tests
     * In rare situations, this may not be possible or applicable to a PR. In those situations:
         1. Note why this could not be done or is not applicable here: 
         2. Add TODOs in the code noting that it requires a test
         3. Add a JIRA task to add the test and link it here: 
- [x] Automated regression test(s) pass

If JIRA tests were used to supplement or replace automated tests:
- [X] JIRA test links: NA
- [X] Justification for using JIRA tests: NA
- [X] JIRA tests have been added to sprint NA

**Reviewer 1:**

Name: @holmesie
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA tests have been run and pass
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree

**Reviewer 2:**

Name: @alexanderelliott121 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] N/A - JIRA tests have been run and pass
- [x] N/A - You agree with the justification for use of JIRA tests or have provided input on why you disagree

### CODE REVIEW

File: app/assets/javascripts/views/measure_value_sets.js.coffee [hunk 1]
```diff
@@ -25,9 +25,16 @@ class Thorax.Views.MeasureValueSets extends Thorax.Views.BonnieView
...
+    if isDirectReference
+      oid_version = ''
+      version = ''
+    else
+      oid_version = _.find(@model.get('value_set_oid_version_objects'), (oid_version) -> oid_version.oid == oid)
+
     if oid_version? && bonnie.valueSetsByOid[oid]?
-      version = oid_version.version
+      if not isDirectReference
+        version = oid_version.version
```
Comment (alexanderelliott121) [inline app/assets/javascripts/views/measure_value_sets.js.coffee]: Is there a reason that this has to be done here, and not in the first isDirectReference else block?

File: app/assets/javascripts/views/measure_value_sets.js.coffee [hunk 2]
```diff
@@ -57,16 +64,36 @@ class Thorax.Views.MeasureValueSets extends Thorax.Views.BonnieView
...
+        # Direct Reference Codes
+        if library.library.codes
+          library.library.codes.def.forEach (code) =>
+            name = code.name
+            display = code.display
+            oid = 'Direct Reference Code'
+            cid = _.uniqueId('c')
+            # Get the guid by looping over bonnie.valueSetByOid
+            for oid, value of bonnie.valueSetsByOid
+              if /-/.test(oid)
```
Comment (alexanderelliott121) [inline app/assets/javascripts/views/measure_value_sets.js.coffee]: If I'm reading this right, for each code in each library it iterates over every entry of bonnie.valueSetsByOid, but in each iteration it only cares about the entries that are direct reference codes (DRCs).

You could do a first pass over bonnie.valueSetsByOid, pushing DRCs to an array then iterate over that in the inner loop.

It might be possible to do better than that, given than terminology will eventually be filtered for duplicates.

**Review by alexanderelliott121 — CHANGES_REQUESTED** [review]
I think there are a few code improvements that I think would be worthwhile to look into.

**Review by alexanderelliott121 — APPROVED** [review]

**Review by holmesie — APPROVED** [review]
